src/skdh/completeness/complete.py

```diff
@@ -0,0 +1,548 @@
+            data_files = data_files + ['Device Battery']
+        if 'Wear Indicator.csv' in os.listdir(subject_folder):
+            data_files = data_files + ['Wear Indicator']
+
+        for measure in data_files:
+            fname = subject_folder + '/' + measure + '.csv'
+            df_raw = pd.read_csv(fname)
+
+            assert 'Time Unix (ms)' in df_raw.keys(), '"Time Unix (ms)" column is missing from file ' + fname
+            assert 'Sampling Frequency (Hz)' in df_raw.keys(), '"Sampling Frequency (Hz)" column is missing from file ' + fname
+            assert measure in df_raw.keys(), measure + ' column is missing from file ' + fname
+            assert df_raw['Time Unix (ms)'].iloc[0] < 10 ** 13 and df_raw['Time Unix (ms)'].iloc[0] > 10 ** 11, \
+                'Unix times are too small or too big, they need to be in ms and should therefore be ~10^12'
+            assert df_raw['Sampling Frequency (Hz)'].dtype in [float, int], 'sfreq must be a number (int/float)'
+
+            if not 'Device ID' in df_raw.keys():
+                df_raw['Device ID'] = 'n/a'
+
+            if not 'Timezone' in df_raw.keys():
+                warnings.warn('No timezone key given, will presume the timezone is EST (-5).')
+                times = [from_unix(df_raw['Time Unix (ms)'], time_unit='ms', utc_offset=-5)]
+                df_raw.set_index(times, inplace=True)
+            else:
+                times = [from_unix(df_raw['Time Unix (ms)'], time_unit='ms', utc_offset=df_raw['Timezone'])]
+                df_raw.set_index(times, inplace=True)
+
+            df_raw['Sampling Frequency (Hz)'] = np.array(1 / df_raw['Sampling Frequency (Hz)'] * 10 ** 9,
+                                                         'timedelta64[ns]')
+            df_raw = df_raw.iloc[np.where(~np.isnan(df_raw[measure]))[0]]
+            df_raw = df_raw.iloc[np.argsort(df_raw['Time Unix (ms)'])]
+
+            if not self.ranges is None:
+                if measure in self.ranges.keys():
+                    df_raw = clean_df(df_raw, measure, self.ranges[measure][0], self.ranges[measure][1])
+            if len(df_raw) > 1:
+                if measure in ['Device Battery', 'Wear Indicator']:
+                    data_dic.update({measure: df_raw})
+                else:
+                    data_dic['Measurement Streams'].update({measure: df_raw})
+
+        return data_dic
+
+    @handle_process_returns(results_to_kwargs=True)
+    def predict(self,
+                data_dic,
+                measures,
+                fpath_output=None,
+                **kwargs):
+        """
+        Compute completeness and save results. Create and save figures if generate_figures is True.
+        """
+
+        assert os.path.isdir(fpath_output), 'fpath_output does not appear to exist'
+
+        super().predict(
+            expect_days=False,
+            expect_wear=False,
+            **kwargs,
+        )
+
+        if self.time_periods is None:
```

Comment:
The time periods are specified by the user as times in numpy datetime format, or as 'daily' in which case it's recording time start plus 24 h, 48h, etc. Since the pipeline works on each subject independently, any batch processing of multiple subjects would just be handled by the user looping through all subjects. Not sure if this is consistent with the rest of SKDH processing but I think so?